### PR TITLE
Renamed `DataLakePathClient::GetAccessControls` to `DataLakePathClient::GetAccessControlList`.

### PR DESCRIPTION
**Pull request by katmsft**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/storage/azure-storage-files-datalake/CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,7 @@
 - Renamed `GetUri` to `GetUrl`.
 - Added `DataLakeLeaseClient`, all lease related APIs are moved to `DataLakeLeaseClient`.
 - Changed lease duration to be `std::chrono::seconds`.
+- Renamed `DataLakePathClient::GetAccessControls` to `DataLakePathClient::GetAccessControl`.
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/CHANGELOG.md:25]: When is this called `AccessControl` and when is it called `AccessControlList`?

Comment (katmsft) [reply sdk/storage/azure-storage-files-datalake/CHANGELOG.md:25]: I don't know, I just align with .NET. https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Files.DataLake/src/DataLakePathClient.cs#L1870

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-datalake/CHANGELOG.md:25]: whatever

Comment (katmsft) [reply sdk/storage/azure-storage-files-datalake/CHANGELOG.md]: Changed to `GetAccessControlList` as agreed.

**Review by Jinming-Hu — APPROVED** [review]

**Pull request merged**